### ISSUE TITLE
Unsatisfactory error message if argument of \colorlet is empty

### ISSUE THREAD

**Issue by wehro** [latex3/xcolor#12]
If the second argument of the `\colorlet` command is empty, the error message is not useful. The `Undefined color` message should be used in this case.
```
documentclass{article}
\usepackage{xcolor}

\begin{document}
\colorlet{mycolor}{}
\end{document}
```

```
Runaway argument?
\def \@@tmp {2}\else \edef \@@tmp {\noexpand \in@ {/./}{/\@@tmp /}}\@@tmp \ETC.
! Paragraph ended before \@car was complete.
```

**Comment by davidcarlisle** [comment]
I think we can guard `\@car` as follows

```
\documentclass{article}
\usepackage{xcolor}

\makeatletter
\def\XC@info#1#2%
 {\XC@edef#2{#1}%
  \expandafter\in@\expandafter>\expandafter{#2}%
  \ifin@\def#2{4}\else\expandafter\in@\expandafter:\expandafter{#2}%
  \ifin@\def#2{3}\else\expandafter\in@\expandafter!\expandafter{#2}%
  \ifin@\def#2{2}\else % now there's no >,:,!
  \if-\expandafter\@car#2\relax\@nil\def#2{2}\else % catches -red
                         %%%%%% add \relax in case #2 expands to empty
  \edef#2{\noexpand\in@{/./}{/#2/}}#2%
  \edef#2{\ifin@0\else1\fi}\fi\fi\fi\fi}
\makeatother
\begin{document}
\colorlet{mycolor}{}
\end{document}
```

**Comment by davidcarlisle** [comment]
this is fixed in  f890f9d
